caddyhttp/fastcgi/fastcgi.go

```diff
@@ -100,6 +102,20 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 			case "OPTIONS":
 				resp, err = fcgiBackend.Options(env)
 			default:
+				// since some chrome releases or quic clients are not sending the Content-Length
+				// in the request headers, caddy should re-calculate content-length by the body and fix it.
+				if contentLength == 0 {
+					var body []byte
+					if r.Body != nil {
+						body, _ = ioutil.ReadAll(r.Body)
```

Comment:
what i mentioned "I just noticed my stable chrome release does not send Content-Length header" above was wrong, it has been sending Content-Length header correctly but I didn't catch it (#1501). However, I think it would be great if caddy can handle the request which does not contain Content-Length header.